src/js/modules/infragistics.ui.widget.js

```diff
@@ -0,0 +1,133 @@
+/*!@license
+ * Infragistics.Web.ClientUI Base Widget <build_number>
+ *
+ * Copyright (c) 2011-<year> Infragistics Inc.
+ * <Licensing info>
+ *
+ * http://www.infragistics.com/
+ *
+ * Depends on:
+ *   jquery-1.9.1.js
+ *   jquery.ui.core.js
+ *   jquery.ui.widget.js
+ *   infragistics.util.js
+ *   infragistics.util.jquery.js
+ */
+
+(function (factory) {
+	if (typeof define === "function" && define.amd) {
+
+		// AMD. Register as an anonymous module.
+		define( [
+			"jquery",
+			"jquery-ui",
+			"./infragistics.util",
+			"./infragistics.util.jquery"
+		], factory );
+	} else {
+
+		// Browser globals
+		factory(jQuery);
+	}
+}
+(function ($) {
+    /*
+		Base igWidget for all Ignite UI widgets.
+	*/
+    $.widget("ui.igWidget", {
+		localeWidgetName: null,
+        options: {
+            /* type="string|object" Set/Get the locale setting for the widget.
+            ```
+                 //Initialize
+                $(".selector").igWidget({
```

Comment:
If we want the same code snippet to appear in all the inherited widget we should use %%WidgetName%% placeholder instead of igWidget